linera-views/src/hashable_wrapper.rs

```diff
@@ -70,25 +70,32 @@ where
         *self.hash.get_mut() = self.stored_hash;
     }
 
+    async fn has_pending_changes(&self) -> bool {
+        if self.inner.has_pending_changes().await {
+            return true;
+        }
+        let hash = self.hash.lock().await;
+        self.stored_hash != *hash
+    }
+
     fn flush(&mut self, batch: &mut Batch) -> Result<bool, ViewError> {
         let delete_view = self.inner.flush(batch)?;
+        let hash = self.hash.get_mut();
         if delete_view {
             let mut key_prefix = self.inner.context().base_key();
             key_prefix.pop();
             batch.delete_key_prefix(key_prefix);
             self.stored_hash = None;
-        } else {
```

Comment:
What is this change about? Can it be in a different PR?

Reply:
Ok, I have a branch for that. I will create a draft PR once the CI is working again.